packages/providers/upload-local/lib/index.js

```diff
@@ -103,3 +103,13 @@ module.exports = {
     };
   },
 };
+
+/**
+ * @param {*} file
+ * @param {boolean} relative
+ * @returns {string}
+ */
+function getFileUrl(file, relative) {
+  const pathname = `/${UPLOADS_FOLDER_NAME}/${file.hash}${file.ext}`;
```

Comment:
I think this implementation could introduce problems. The url you assign to a file will be stored in db, so if you assign an absolute path and you change your project url:
- Changing the port
- Changing the proxy you have behind it.

That would break all links you have stored in the database.

I don't think this should be done at the provider level, as it can break those ways. We could explore other ways, such as intercepting all media attributes, seeing if the path is relative and updating them to be absolute, something like:


```

const changePathsToAbsolute = async () => {

  // You can extend service in strapi, meaning you can intercept the data and make changes
  // before it gets to the admin UI or the Content API.
  // This here would intercept media files that are relative and change the string to be absolute
  // (This feature will be merged in 4.9.0, in a week from today)
  strapi.container
    .get('services')
    .extend('plugin::content-manager.entity-manager', (entityManager) => {
      const mapEntity = async (entity, uid) => {
        // Map entity is a method that is called before returning the data to the admin
        // Here you can extend it to modify the entities as you like
        const mappedEntity = await entityManager.mapEntity(entity, uid);
        // We would have to implement this
        return updatePathsToAbsolute(mappedEntity, uid);
      };
   
      return { ...entityManager, mapEntity };
    });
};
```

Would you be up to follow on this? :)

Reply:
This would be the ideal solution. In DB links are stored relative/absolute (based on provider) in API links are always absolute.

Reply:
agree, the way we store the url is based on the provider. Just that the way we store the url we could approach it in a way that can be more useful for people who want a different behaviour.

Also, another possibility is to create a custom local provider that do stores the url this way.
I just think we shouldn't introduce a fix for this that might not cover the majority of use cases when people wants absolute paths using the local provider.

Reply:
Ok. Then I will refactor this PR to just add some tests to check if URL is relative

Reply:
@Marc-Roig

Reply:
👍 sounds good to me, thank you!!